README.md

```diff
@@ -22,173 +18,177 @@ Currently, our SDK supports interactions with our application-specific rollup ba
 
 ## Documentation
 
-See the [developer guides](https://docs.x.immutable.com) for information on building on Immutable X.
+See our [developer documentation](https://docs.x.immutable.com) for information on building on Immutable X.
 
-See the [API reference documentation](https://docs.x.immutable.com/reference) for more information on our API's.
+See our [API reference](https://docs.x.immutable.com/reference) for more information on our APIs.
 
 ## Installation
 
-## Add this NPM package to your project
-
 ```sh
 npm install @imtbl/core-sdk --save
 # or
 yarn add @imtbl/core-sdk
 ```
 
-## Setup local developer environment
+## Initialization
 
-```sh
-# install dependencies
-yarn install
-yarn build
-# run tests
-yarn test
+Initialize the Core SDK client with the network on which you want your application to run (see [all networks available](./src/config/config.ts)):
+| Param | Description |
+| -- | -- |
+| `Config.SANDBOX` | The default test network (currently, it is Goërli) |
+| `Config.ROPSTEN` | Ropsten test network (to be deprecated soon) |
+| `Config.PRODUCTION` | Ethereum network |
+
+```ts
+import { ImmutableX, Config } from '@imtbl/core-sdk';
+
+const config = Config.SANDBOX; // Or PRODUCTION or ROPSTEN
+const client = new ImmutableX(config);
 ```
 
-Check out how the [Release Process](https://github.com/immutable/imx-core-sdk/#release-process) works
+## Get data (on assets, orders, past transactions, etc.)
 
-## Usage
+These methods allow you to read data about events, transactions or current state on Immutable X (layer 2). It does not require any user authentication because no state is being changed.
+
+Examples of the types of data that is typically retrieved include:
+
+- Assets or details of a particular asset
+- Token balances for a particular user
+- Orders or details about a particular order
+- Historical trades and transfers
+
+### Examples
+
+#### Get all collections and get assets from a particular collection:
 
 ```ts
-import { AlchemyProvider } from '@ethersproject/providers';
-import { Wallet } from '@ethersproject/wallet';
-import { Utils } from '@imtbl/core-sdk';
+const listCollectionsResponse = await client.listCollections({
+  pageSize: 2,
+});
+
+const collection = listCollectionsResponse.data.result[0];
+
+const collectionAssetsResponse = await client.listAssets({
+  collection: collection.address,
+  pageSize: 10,
+});
 ```
 
-#### WalletConnection
+## Operations requiring user signatures
 
-WalletConnection is a top level connector which contains all the required resources (eg. L1 and L2 signers) to enable usage of the Core SDK workflows.
+There are two types of operations requiring user signatures:
+1. Transactions that update blockchain state
+2. Operations that Immutable X require authorization for
+
+In order to get user signatures, applications need to [generate "signers"](#how-do-applications-generate-and-use-signers).
+
+#### What are transactions that update blockchain state?
```

Comment:
Can you add a newline here? (for code sake, not sure if it makes a difference on the formatting in preview)